libraries/c_sdk/standard/https/src/iot_https_client.c

```diff
@@ -2464,6 +2344,56 @@ IotHttpsReturnCode_t IotHttpsClient_Init( void )
 
+    /* Allocate the dispatch queue. */
+    #if IOT_HTTPS_DISPATCH_USE_STATIC_MEMORY == 1
+        dispatchQueue = xQueueCreateStatic( IOT_HTTPS_DISPATCH_QUEUE_SIZE,
+                                            sizeof( _httpsRequest_t * ),
+                                            dispatchQueueStorageBuffer,
+                                            &dispatchQueueBuffer );
+    #else
+        dispatchQueue = xQueueCreate( IOT_HTTPS_DISPATCH_QUEUE_SIZE, sizeof( _httpsRequest_t * ) );
+    #endif
+
+    if( dispatchQueue == NULL )
+    {
+        /* Queue was not created and must not be used. */
+        IotLogError( "Failed to allocate resources for dispatch queue.", status );
+        HTTPS_SET_AND_GOTO_CLEANUP( IOT_HTTPS_INTERNAL_ERROR );
+    }
+
+    /* Start tasks that send requests from the dispatch queue. */
+    for( dispatchTaskIndex = 0; dispatchTaskIndex < IOT_HTTPS_DISPATCH_TASK_COUNT; ++dispatchTaskIndex )
+    {
+        #if IOT_HTTPS_DISPATCH_USE_STATIC_MEMORY == 1
+            httpsDispatchTask[ dispatchTaskIndex ] = xTaskCreateStatic( _dispatchTaskRoutine,
+                                                                        "iot_thread",
+                                                                        IOT_HTTPS_DISPATCH_TASK_STACK_SIZE,
+                                                                        NULL,
+                                                                        IOT_HTTPS_DISPATCH_TASK_PRIORITY,
+                                                                        dispatchTaskStack,
+                                                                        &dispatchTaskBuffer );
+
+            if( httpsDispatchTask[ dispatchTaskIndex ] == NULL )
+            {
+                IotLogError( "Failed to allocate static memory for request task.", status );
+                HTTPS_SET_AND_GOTO_CLEANUP( IOT_HTTPS_INTERNAL_ERROR );
+            }
+        #else /* ifdef IOT_HTTPS_DISPATCH_USE_STATIC_MEMORY */
+            taskCreationResult = xTaskCreate( _dispatchTaskRoutine,
+                                              "iot_thread",
+                                              IOT_HTTPS_DISPATCH_TASK_STACK_SIZE,
+                                              NULL,
+                                              IOT_HTTPS_DISPATCH_TASK_PRIORITY,
+                                              &httpsDispatchTask[ dispatchTaskIndex ] );
+
+            if( taskCreationResult != pdPASS )
+            {
+                IotLogError( "Failed to allocate dynamic memory for request task.", status );
+                HTTPS_SET_AND_GOTO_CLEANUP( IOT_HTTPS_INTERNAL_ERROR );
+            }
+        #endif /* ifdef IOT_HTTPS_DISPATCH_USE_STATIC_MEMORY */
+    }
+
 /* This code prints debugging information and is, therefore, compiled only when
  * log level is set to IOT_LOG_DEBUG. */
     #if ( LIBRARY_LOG_LEVEL == IOT_LOG_DEBUG )
```

Comment:
HTTPS_FUNCTION_EXIT_NO_CLEANUP(); might need to change to deleting the queue if it is created and deleting any non null tasks created, upon error.

Reply:
Thanks, missed that.